cucumber/index.d.ts

```diff
@@ -70,5 +70,139 @@ declare namespace cucumber {
 		Around(code: AroundCode):void;
 		setDefaultTimeout(time:number): void;
 		registerHandler(handlerOption:string, code:(event:any, callback:CallbackStepDefinition) =>void): void;
+		registerListener(listener: EventListener): void;
+	}
+
+	export class EventListener {
+		hear(event: events.Event, callback: ()=>void): void;
+		hasHandlerForEvent(event: events.Event): boolean;
+		buildHandlerNameForEvent(event: events.Event): string;
+		getHandlerForEvent(event: events.Event): EventHook;
+		buildHandlerName(shortName: string): string;
+		setHandlerForEvent(shortName: string, handler: EventListener): void;
+	}
+
+	export function Listener(): EventListener;
+
+	export namespace events {
+
+		interface Event {
+			getName(): string;
+			getPayloadItem(name: string): EventPayload;
+		}
+
+		interface EventPayload {
```

Comment:
Could you provide an example of using the `EventPayload` type?
To clarify, an empty interface really is equivalent to `{}` and doesn't "mark" anything. So `const foo: EventPayload = 0;` would compile without error.
If you need to specify that only the things you've designated here as `EventPayload`s qualify, you could use a union type: `type EventPayload = FeaturesPayload | FeatueresResultPayload | ... | StepResultPayload`.